### PR TITLE
Add unit- & end to end test setup in frontend

### PR DESCRIPTION
**Pull request by FrankStrieter**
## Proposed change
- deleted most of the boilerplate tests
  -  it seems to a developer, that the codebase is fully tested, but it is not.
- move from karma to jest
- move from protractor to cypress

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Other (please explain)
  - add and fixes test setup in fronted

Open tasks:
- [x] add test runs to ci pipeline
- [x] check code format

### CODE REVIEW

**Comment by shamoon** [comment]
This is really great and something I had hoped to do, thank you so much for digging into it.

**Review by shamoon — COMMENTED** [review]
This changes the package-lock file back to version 1, I assume we dont really want that. My npm updated it (back) to v2, would you like me to commit that here?

**Comment by shamoon** [comment]
Also are you OK with me adding more tests to this?

**Comment by FrankStrieter** [comment]
sure, but I do not want to have this as a long-running merge request. wen can add more and more tests in the next days and weeks.

I'll check this with the npm stuff. maybe I was on another node version due to the regular work.

For now I need to fix the code style issues coming from my IDE configured prettier.

**Comment by shamoon** [comment]
> sure, but I do not want to have this as a long-running merge request. wen can add more and more tests in the next days and weeks.

Yes agreed, whenever it's ready we can merge this. Also since this is on `main` some of the frontend stuff is outdated so some tests will need to come after we merge this back into `dev` anyway.

Thanks again, Im really excited to have this off the ground.

File: src-ui/cypress/integration/documents-list.spec.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+describe('documents-list', () => {
+  beforeEach(() => {
+    cy.intercept('http://localhost:8000/api/documents/*', {
+      fixture: 'documents/documents.json',
+    });
+    cy.intercept('http://localhost:8000/api/documents/1/thumb/', {
+      fixture: 'documents/lorem-ipsum.png',
+    });
+
+    cy.visit('/documents');
```
Comment (lippoliv) [inline src-ui/cypress/integration/documents-list.spec.ts:10]: I think `cy.visit` should be moved to the test, so that this spec is resilient to future modifications / refactorings?

File: src-ui/src/app/app.component.spec.ts [hunk 2]
```diff
@@ -1,35 +0,0 @@
...
-    expect(app.title).toEqual('paperless-ui');
-  });
-
-  it('should render title', () => {
-    const fixture = TestBed.createComponent(AppComponent);
-    fixture.detectChanges();
-    const compiled = fixture.nativeElement;
-    expect(compiled.querySelector('.content span').textContent).toContain('paperless-ui app is running!');
-  });
-});
```
Comment (lippoliv) [inline src-ui/src/app/app.component.spec.ts:35]: I think these specs should not be removed. cypress for sure is usefull for integration/e2e testing but this must not replace unit tests. (applies to all `*.spec.ts` removals)

File: src-ui/src/app/components/common/edit-dialog/edit-dialog.component.ts [hunk 3]
```diff
@@ -2,104 +2,111 @@ import { Directive, EventEmitter, Input, OnInit, Output } from '@angular/core';
...
+        this.networkActive = false;
+      }
+    );
   }
 
   cancel() {
-    this.activeModal.close()
+    this.activeModal.close();
   }
 }
```
Comment (lippoliv) [inline src-ui/src/app/components/common/edit-dialog/edit-dialog.component.ts:112]: There is no change, right? I would preffer to revert this changes so we don't mess up the git history?

File: src-ui/src/app/components/common/input/check/check.component.ts [hunk 4]
```diff
@@ -1,22 +1,21 @@
...
+  styleUrls: ['./check.component.scss'],
 })
 export class CheckComponent extends AbstractInputComponent<boolean> {
-
   constructor() {
-    super()
+    super();
   }
-
 }
```
Comment (lippoliv) [inline src-ui/src/app/components/common/input/check/check.component.ts:21]: There is no change, right? I would preffer to revert this changes so we don't mess up the git history?

**Review by lippoliv — CHANGES_REQUESTED** [review]
I think bringing cypress to the project can be a benefit. But in my opinion it should not replace the unit tests for the components.

File: src-ui/src/app/components/common/edit-dialog/edit-dialog.component.ts (hunk 3 above)
Comment (FrankStrieter) [reply src-ui/src/app/components/common/edit-dialog/edit-dialog.component.ts:113]: sure. my installed prettier from the IDe tried to solve some issues :-D

**Comment by FrankStrieter** [comment]
> I think bringing cypress to the project can be a benefit. But in my opinion it should not replace the unit tests for the components.

I absolute agree with you. This pull request does only configure the tets and make them run. So that we can go ahead and to testing on the frontend.

**Comment by FrankStrieter** [comment]
> I think these specs should not be removed. cypress for sure is usefull for integration/e2e testing but this must not replace unit tests. (applies to all *.spec.ts removals)

I removed all boilerplate-tests that are useless and tests nothing and fails due to service dependencies. It will be a mess and a huge pull request to fix all the deleted tests in one shot. we can start now and add real unit tests.

File: src-ui/cypress/integration/documents-list.spec.ts (hunk 1 above)
Comment (shamoon) [reply src-ui/cypress/integration/documents-list.spec.ts:10]: yes, agree

**Comment by shamoon** [comment]
Why is this adding semicolons to some of the files? Our prettier settings not include them, see: https://github.com/paperless-ngx/paperless-ngx/blob/dev/.prettierrc

**Comment by FrankStrieter** [comment]
> Why is this adding semicolons to some of the files? Our prettier settings not include them, see: https://github.com/paperless-ngx/paperless-ngx/blob/dev/.prettierrc

my IDE changed this. I am still solve these issues. please hold on.

File: .github/workflows/project-actions.yml [hunk 5]
```diff
@@ -8,6 +8,9 @@ on:
     types:
       - opened
       - reopened
+    branches:
+      - main
+      - dev
```
Comment (shamoon) [inline .github/workflows/project-actions.yml:13]: Sorry I also think this shouldn't be in here, will cause a merge conflict when goes back to `main`

Comment (FrankStrieter) [reply .github/workflows/project-actions.yml]: done

**Comment by lippoliv** [comment]
> > I think these specs should not be removed. cypress for sure is usefull for integration/e2e testing but this must not replace unit tests. (applies to all *.spec.ts removals)
> 
> I removed all boilerplate-tests that are useless and tests nothing and fails due to service dependencies. It will be a mess and a huge pull request to fix all the deleted tests in one shot. we can start now and add real unit tests.

Got it. Sorry I did not check each and every spec file, "murphys law" let me select app component which at least checks some title. 

I agree, if the tests do not work, it looks like "false safety feeling" (e.g. "nice, there are specs so this app ui is covered by tests"). So removing them and then step by step bringing them back is probably a good idea.

**Review by shamoon — APPROVED** [review]
